usaspending_api/common/recipient_lookups.py

```diff
@@ -1,49 +1,41 @@
 from usaspending_api.recipient.models import RecipientLookup
 
 
-def obtain_recipient_uri(
-        recipient_name,
-        recipient_unique_id,
-        parent_recipient_unique_id=None,
-        is_parent_recipient=False):
-    # No need to create a hash when the parent recipient does not exist
-    if is_parent_recipient and not recipient_unique_id:
+def obtain_recipient_uri(recipient_name, recipient_unique_id, parent_recipient_unique_id, is_parent_recipient=False):
+    """ Return a valid string to be used for api/v2/recipient/duns/<recipient-hash>/ (or None)
+
+        Keyword Arguments:
+        recipient_name -- Legal Entity Name from the record
+        recipient_unique_id -- DUNS from the record
+        parent_recipient_unique_id -- parent DUNS from the record
+        is_parent_recipient -- boolean flag to force the recipient level to be "P" (default False)
+            By the nature of transaction records, the listed recipient can only be "R" or "C"
+            This flag is for the parent recipient link (as appropriate)
+
+        Return example string: 11fcdf15-3490-cdad-3df4-3b410f3d9b20-C
+
+    """
+    recipient_hash = None
+    if (is_parent_recipient and not recipient_unique_id) or not (recipient_unique_id or recipient_name):
         return None
-    elif not recipient_unique_id:
-        return create_recipient_uri_without_duns(
-            recipient_name,
-            recipient_unique_id,
-            parent_recipient_unique_id,
-            is_parent_recipient
-        )
-    return fetch_recipient_uri_with_duns(recipient_unique_id, parent_recipient_unique_id, is_parent_recipient)
-
-
-def create_recipient_uri_without_duns(
-        recipient_name,
-        recipient_unique_id,
-        parent_recipient_unique_id=None,
-        is_parent_recipient=False):
-    recipient_hash = generate_missing_recipient_hash(recipient_name, recipient_unique_id)
-    recipient_level = obtain_recipient_level({
-        "duns": recipient_unique_id,
-        "parent_duns": parent_recipient_unique_id,
-        "is_parent_recipient": is_parent_recipient
-    })
-    return combine_recipient_hash_and_level(recipient_hash, recipient_level)
-
+    elif recipient_unique_id:
+        recipient_hash = fetch_recipient_hash_using_duns(recipient_unique_id)
```

Comment:
This logic could be arranged more cleanly.

Ex: if recpient_unique_id return fetch...
   if parent return none
   if not recipient_name return none

There's other ways to do it but it looks like you don't need that recipient_hash value to be used again if recipient_unique_id is present

Reply:
@BrianZito If I understand your example, there is an assumption that a pre-calculated recipient hash will always be found by the DUNS. This isn't true 100% of the time, especially in testing environments

Reply:
Ok, I didn't take that into account. In that case, I would group the attempt to get hash from duns with the lines below that catch substitute with generate_missing_recipient_hash (which I assume never returns none), and put the special cases above by themselves. The recipient_hash variable is really only used to mark if the dun fetch worked, which confused me the first few times I looked at this.

Reply:
new Ex:
if yada yada case 1:
return none because we don't have enough info

duns_fetch = try_to_fetch_on_duns
if not duns_fetch:
do backup thing

Reply:
In that recent example it is possible to search RecipientLookup with a `null` in `recipient_unique_id`, which won't provide desirable behavior.

Adding the if statement back in, it's exactly like the code in the PR